tests/measurements/test_measurements.py

```diff
@@ -0,0 +1,481 @@
+        meas_proc = q.queue[-1]
+        assert isinstance(meas_proc, MeasurementProcess)
+        assert meas_proc.return_type == return_type
+
+        assert q.get_info(A) == {"owner": meas_proc}
+        assert q.get_info(meas_proc) == {"owns": (A)}
+
+    def test_annotating_tensor_hermitian(self, stat_func, return_type):
+        """Test that the return_type related info is updated for a measurement
+        when called for an Hermitian observable"""
+
+        mx = np.array([[1, 0], [0, 1]])
+
+        with AnnotatedQueue() as q:
+            Herm = qml.Hermitian(mx, wires=[1])
+            stat_func(Herm)
+
+        assert q.queue[:-1] == [Herm]
+        meas_proc = q.queue[-1]
+        assert isinstance(meas_proc, MeasurementProcess)
+        assert meas_proc.return_type == return_type
+
+        assert q.get_info(Herm) == {"owner": meas_proc}
+        assert q.get_info(meas_proc) == {"owns": (Herm)}
+
+    @pytest.mark.parametrize(
+        "op1,op2",
+        [
+            (qml.PauliY, qml.PauliX),
+            (qml.Hadamard, qml.Hadamard),
+            (qml.PauliY, qml.Identity),
+            (qml.Identity, qml.Identity),
+        ],
+    )
+    def test_annotating_tensor_return_type(self, op1, op2, stat_func, return_type):
+        """Test that the return_type related info is updated for a measurement
+        when called for an Tensor observable"""
+        with AnnotatedQueue() as q:
+            A = op1(0)
+            B = op2(1)
+            tensor_op = A @ B
+            stat_func(tensor_op)
+
+        assert q.queue[:-1] == [A, B, tensor_op]
+        meas_proc = q.queue[-1]
+        assert isinstance(meas_proc, MeasurementProcess)
+        assert meas_proc.return_type == return_type
+
+        assert q.get_info(A) == {"owner": tensor_op}
+        assert q.get_info(B) == {"owner": tensor_op}
+        assert q.get_info(tensor_op) == {"owns": (A, B), "owner": meas_proc}
+
+    @pytest.mark.parametrize(
+        "op1,op2",
+        [
+            (qml.PauliY, qml.PauliX),
+            (qml.Hadamard, qml.Hadamard),
+            (qml.PauliY, qml.Identity),
+            (qml.Identity, qml.Identity),
+        ],
+    )
+    def test_queueing_tensor_observable(self, op1, op2, stat_func, return_type):
+        """Test that if the constituent components of a tensor operation are not
+        found in the queue for annotation, they are not queued or annotated."""
+        A = op1(0)
+        B = op2(1)
+
+        with AnnotatedQueue() as q:
+            tensor_op = A @ B
+            stat_func(tensor_op)
+
+        assert len(q._queue) == 2  # pylint: disable=protected-access
+
+        assert q.queue[0] is tensor_op
+        meas_proc = q.queue[-1]
+        assert isinstance(meas_proc, MeasurementProcess)
+        assert meas_proc.return_type == return_type
+
+        assert q.get_info(tensor_op) == {"owns": (A, B), "owner": meas_proc}
+
+
+@pytest.mark.parametrize("stat_func", [expval, var, sample])
+class TestBetaStatisticsError:
```

Comment:
Not exactly "beta" if its been the behaviour for two years.  I think this test should just merged into another test class.

Reply:
Done!